### PR TITLE
Testing: added common ancestor to mock commands

### PR DESCRIPTION
**Pull request by MBoldyrev**
Signed-off-by: Mikhail Boldyrev <miboldyrev@gmail.com>

### Description of the Change

Specific mock commands now inherit a common base class that provides a helper type holding the interface they implement.

### Benefits

This is needed for upcoming Executor ITF.

### CODE REVIEW

File: test/module/shared_model/command_mocks.hpp [hunk 1]
```diff
@@ -36,44 +36,59 @@ namespace shared_model {
...
+
+    template <typename T>
+    struct SpecificMockCommand : public T, public SpecificMockCommandBase {
+      using SpecificCommand = T;
+    };
+
     struct MockAddAssetQuantity
-        : public shared_model::interface::AddAssetQuantity {
+        : public SpecificMockCommand<
+              shared_model::interface::AddAssetQuantity> {
```
Comment (igor-egorov) [inline test/module/shared_model/command_mocks.hpp]: You are already in the namespace
```suggestion
              AddAssetQuantity> {
```

Comment (igor-egorov) [reply test/module/shared_model/command_mocks.hpp]: Please apply the same fix for the rest commands.

File: test/module/shared_model/command_mocks.hpp [hunk 2]
```diff
@@ -32,48 +32,54 @@ using testing::Return;
...
       MOCK_CONST_METHOD0(get, const Command::CommandVariantType &());
     };
 
-    struct MockAddAssetQuantity
-        : public shared_model::interface::AddAssetQuantity {
+    struct SpecificMockCommandBase {};
+
+    template <typename T>
+    struct SpecificMockCommand : public T, public SpecificMockCommandBase {
+      using SpecificCommand = T;
```
Comment (lebdron) [inline test/module/shared_model/command_mocks.hpp]: Please use 
https://github.com/hyperledger/iroha/blob/b01d1f8158ab6f24e1d90443c47118f4368fc022/shared_model/interfaces/base/model_primitive.hpp#L28 or https://github.com/hyperledger/iroha/blob/b01d1f8158ab6f24e1d90443c47118f4368fc022/shared_model/interfaces/base/noncopyable_model_primitive.hpp#L22 instead

Comment (MBoldyrev) [reply test/module/shared_model/command_mocks.hpp]: I removed this alias, but I still need the common ancestor class to distinguish a command from other `Model`s.

**Comment by MBoldyrev** [comment]
@lebdron proposed a way to achieve the needed behaviour in Executor ITF without this change, so it is no longer needed.

**Pull request closed without merging**